### PR TITLE
adding initial language model template work

### PR DESCRIPTION
**Pull request by mgerchick**
@EziOzoani @nateraw initial work on a language model template

### CODE REVIEW

File: modelcards/language_model_template.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+- **Compute Region:** {{ compute_region }}
+- **Carbon Emitted** *(Power consumption x Time x Carbon produced based on location of power grid)*:
+
+### Citation Information
+
+```bibtex
+@inproceedings{...,
+  year={2020}
+}
+```
```
Comment (nateraw) [inline modelcards/language_model_template.md:171]: Probably want to parametrize this as well. Similarly, the `datasets` repo has a `"CITATION"` kwarg within the dataset builder scripts. Folks can define it if they choose. If not, leave it as it is here.

Comment (nateraw) [inline modelcards/language_model_template.md:171]: Probably want to parametrize this as well. Similarly, the `datasets` repo has a `"CITATION"` kwarg within the dataset builder scripts. Folks can define it if they choose. If not, leave it as it is here.

**Review by nateraw — COMMENTED** [review]
We gotta do something about making this actually usable in the lib (but I can help with that). We'll need to update `template_path` to `template_name_or_path`, and allow for folks to specify 'default', 'language-modeling', etc. or something. Or maybe we just look at the provided card data and if language-modeling is in the tags, we use this instead. 

Another thought...This card is _very_ `transformers` specific. Maybe we should have tasks/lib mapping as we build this out (RE: #21 ). I wouldn't bother with this right now, but it's just a thought. Little confusing to go from default card that's generic -> language modeling card that only really applies to models trained with `transformers`. 

As mentioned offline, I think it would be best if we take all the things that aren't LM specific and move them into the default card. The default card should be identical to this one, just without the LM stuff.

File: modelcards/language_model_template.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
+---
+{{card_data}}
+---
+
+# {{ model_id | default("MyModelName", true)}}
+
+## Model Details
+
+{{ model_id | default("MyModelName", true)}} is a{% if child_type|length %} {{ child_type }}{% endif %} language model{% if model_tasks|length %}that can be used for {% for task in model_tasks %}{{ task }}{% if not loop.last %}, {% endif %}{% endfor %}{% endif %}. {% if parent_model|length %}Users of this model card should also consider information about the design, training, and limitations of {{ parent_model }}{% endif %}.
```
[the remark this answers is not in the document]
Comment (mgerchick) [reply modelcards/language_model_template.md:9]: for example:

from modelcards import ModelCard, CardData

repo_id = "Marissa/model-card-testing"

card_data = CardData(
    license='mit',
    language='en',
)

card = ModelCard.from_template(
    card_data,
    template_path = 'language_model_template.md', 
    model_id=repo_id.split('/')[-1], 
    model_description="testing description", 
    model_type="testing type", 
    model_tasks=["text generation"],
    child_type="distilled", 
    parent_model="gpt2", 
    authors=["author1", "author2"])

File: modelcards/language_model_template.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+### Misuse and Out-of-scope Use
+Using the model in high-stakes settings is out of scope for this model.  The model is not designed for critical decisions nor uses with any material consequences on an individual's livelihood or wellbeing. The model outputs content that appears factual but is not correct.
+
+### Limitations
+
+**CONTENT WARNING: Readers should be aware this section contains content that is disturbing, offensive, and can propogate historical and current stereotypes.**
+
+Significant research has explored bias and fairness issues with models for language generation (see, e.g., [Sheng et al. (2021)](https://aclanthology.org/2021.acl-long.330.pdf) and [Bender et al. (2021)](https://dl.acm.org/doi/pdf/10.1145/3442188.3445922)). This model also has persistent bias issues, as highlighted in these demonstrative examples below. Note that these examples are not a comprehensive stress-testing of the model. Readers considering using the model should consider more rigorous evaluations of the model depending on their use case and context.
+
+{% if child_type=='distilled' %}
```
Comment (nateraw) [inline modelcards/language_model_template.md:94]: Would be really great if we could:
1. validate that the given `parent_model` is a model existing on the hub
2. If it exists, link directly to it here, citing something along the lines of "this model inherits limitations and biases from <parent model>, so you should read about those here: <embedded parent model link>.

File: modelcards/language_model_template.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+{{card_data}}
+---
+
+# {{ model_id | default("MyModelName", true)}}
+
+## Model Details
+
+{{ model_id | default("MyModelName", true)}} is a{% if child_type|length %} {{ child_type }}{% endif %} language model{% if model_tasks|length %}that can be used for {% for task in model_tasks %}{{ task }}{% if not loop.last %}, {% endif %}{% endfor %}{% endif %}. {% if parent_model|length %}Users of this model card should also consider information about the design, training, and limitations of {{ parent_model }}{% endif %}.
+
+- **Developed by:** {% for author in authors %}{{ author }}{% if not loop.last %}, {% endif %}{% endfor %}
```
[the remark this answers is not in the document]
Comment (mgerchick) [reply modelcards/language_model_template.md:11]: yep!

File: modelcards/language_model_template.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+- **Model type:** {{ model_type }}
+- **Language(s):** {{ language }} # not working right now
```
[the remark this answers is not in the document]
Comment (mgerchick) [reply modelcards/language_model_template.md:13]: yes good point

File: modelcards/language_model_template.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+Preprocessing, hardware used, hyperparameters...
+
+## Evaluation Results
+
+This model achieves the following results:
+
+NOTE: This will need customization.
+
+{% if model_tasks|length and model_tasks[0] == 'text generation' %}
+| Dataset  | LAMBADA | LAMBADA | CBT-CN | CBT-NE | WikiText2 | PTB    | enwiki8 | text8  | WikiText103 | 1BW   |
```
[the remark this answers is not in the document]
Comment (mgerchick) [reply modelcards/language_model_template.md:143]: this is a good question! i think you're right that they're not always applicable, and I see the value in doing an empty table and then populating it based on their input. I think perhaps as we build out the template this question may answer itself as well!

File: modelcards/language_model_template.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+
+## Environmental Impact
+
+You can estimate carbon emissions using the [Machine Learning Impact calculator](https://mlco2.github.io/impact#compute) presented in [Lacoste et al. (2019)](https://arxiv.org/abs/1910.09700)
+
+- **Hardware Type:** {{ hardware_type }}
+- **Hours used:** {{ hours_training }}
+- **Cloud Provider:** {{ cloud_provider }}
+- **Compute Region:** {{ compute_region }}
+- **Carbon Emitted** *(Power consumption x Time x Carbon produced based on location of power grid)*:
```
[the remark this answers is not in the document]
Comment (mgerchick) [reply modelcards/language_model_template.md:163]: good point!

File: modelcards/language_model_template.md (hunk 1 above)
Comment (mgerchick) [reply modelcards/language_model_template.md:171]: i was wondering about how to parametrize this! great idea

File: modelcards/language_model_template.md (hunk 3 above)
Comment (mgerchick) [reply modelcards/language_model_template.md:94]: yes!!! also doing this in the model details section would be great too

**Comment by adrinjalali** [comment]
Should this code not be in `transformers` rather than this library?

**Comment by nateraw** [comment]
> Should this code not be in `transformers` rather than this library?

Yea, It feels too specific to be here, especially right now. I would prefer we only update the default template for now with whatever suggestions the science team has, and leave downstream templates to outside codebases/apps.

**Comment by nateraw** [comment]
Going to close this one for now and will move forward with #32 . More changes coming to that PR from convos offline.

**Pull request closed without merging**